### ISSUE TITLE
Pysa check always fail when build Pyre(and Pysa)

### ISSUE THREAD

**Issue by WangGithubUser** [facebook/pyre-check#752]
I found that, pysa check was always fail since ea90f020110a565851da045cff796224c6443a0e
If you read the message after pysa fail, you will find that it always fails at a same problem.
It always log that`opam: --assume-depexts was added in version 2.1 of the opam CLI, but version 2.0 has been requested, which is older.`(Usually at line 36, but I'm not sure)
However, we can bump the version of `opam CLI` to 2.1 to fix this issue.

**Comment by arthaud** [comment]
Hi @WangGithubUser, thanks for reaching out.

Could you try changing `opam-version: "2.0"` into `opam-version: "2.1"` in `source/hack_parallel.opam` and see if you still get the same error?

**Comment by WangGithubUser** [comment]
@arthaud Yes, I still get same error.
> Could you try changing into in and see if you still get the same error?`opam-version: "2.0"``opam-version: "2.1"``source/hack_parallel.opam`

However, I found the core problem.
The check ran the following commands(in scripts/setup.py)
```python
def opam_environment_variables(self) -> Dict[str, str]:
    LOG.info("Activating opam")
    opam_env_result = self.run(
        [
            "opam",
            "env",
            "--yes",
            "--switch",
            self.switch_name(),
            "--root",
            self.opam_root.as_posix(),
            "--set-root",
            "--set-switch",
        ]
    )
    ...
```
It means, it ran `opam env ...(args)`
If you see the help doc of `opam env`, you will get this:
```
       --cli=MAJOR.MINOR (absent=2.1)
           Use the command-line interface syntax and semantics of MAJOR.MINOR.
           Intended for any persistent use of opam (scripts, blog posts,
           etc.), any version of opam in the same MAJOR series will behave as
           for the specified MINOR release. The flag was not available in opam
           2.0, so to select the 2.0 CLI, set the OPAMCLI environment variable
           to 2.0 instead of using this parameter.
```
It means, once you ran `opam env -some_args`, the OPAMCLI will set as 2.0
I try to solve this problem at #753

**Comment by WangGithubUser** [comment]
This issue has resolved in b9386eb42d19eaf51c052d9c52d5044d9d10b235, I will close this PR
